pkg/operator/controller/externaldns/controller.go

```diff
@@ -88,16 +93,16 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 
 	if _, _, err := r.ensureExternalDNSNamespace(ctx, r.config.Namespace); err != nil {
-		// Return if the externalDNS namespace cannot be created since
-		// resource creation in a namespace that does not exist will fail.
-		return result, fmt.Errorf("failed to ensure externalDNS namespace: %v", err)
+		return requeueResult, fmt.Errorf("failed to ensure externalDNS namespace: %w", err)
 	}
 
-	haveServiceAccount, _, err := r.ensureExternalDNSServiceAccount(ctx, r.config.Namespace, externalDNS)
+	_, sa, err := r.ensureExternalDNSServiceAccount(ctx, r.config.Namespace, externalDNS)
 	if err != nil {
-		return result, fmt.Errorf("failed to ensure externalDNS service account: %w", err)
-	} else if !haveServiceAccount {
-		return result, fmt.Errorf("failed to get externalDNS service account: %w", err)
```

Comment:
Seems like it makes sense to keep this check.

Reply:
I don't know, just to return a different error message? `failed to get` vs `failed to ensure`?

Reply:
If the serviceaccount isn't there, `sa` will be nil, and `desiredExternalDNSDeployment` will deference a nil pointer.  (Granted, it's extremely unlikely that `ensureExternalDNSServiceAccount` will return false and a nil error value, but it could happen if it does a `Create` and a subsequent `Get` and the `Get` gets a non-found response.  Anyway, we get a Boolean return value here, so it makes sense to check it.)

Reply:
Yes, you are right, good catch, thank you!